src/foam/dao/OptionDAO.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'OptionDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `
+    OptionDAO allows the user to pass a delegate to either the default delegate (OptionDAO.delegate) or
```

Comment:
I find you documentation a little unclear.  How about: 

This DAO supports branching to another DAO when the predicate evaluates to true.